src/sentry/tasks/delete_seer_grouping_records.py

```diff
@@ -50,10 +51,23 @@ def delete_seer_grouping_records_by_hash(
         return
 
     len_hashes = len(hashes)
-    end_index = min(last_deleted_index + BATCH_SIZE, len_hashes)
-    call_seer_to_delete_these_hashes(project_id, hashes[last_deleted_index:end_index])
-    if end_index < len_hashes:
-        delete_seer_grouping_records_by_hash.apply_async(args=[project_id, hashes, end_index])
+    if len_hashes <= BATCH_SIZE:  # Base case
+        call_seer_to_delete_these_hashes(project_id, hashes)
+    else:
+        if last_deleted_index != 0:
+            # This tracks which tasks are still being scheduled with the whole list of hashes
+            metrics.incr(
+                "grouping.similarity.delete_seer_grouping_records_by_hash.batch_size_exceeded",
+                sample_rate=options.get("seer.similarity.metrics_sample_rate"),
+            )
+
+        # Iterate through hashes in chunks and schedule a task for each chunk
+        # There are tasks passing last_deleted_index, thus, we need to start from that index
+        # Eventually all tasks will pass 0
+        for i in range(last_deleted_index, len_hashes, BATCH_SIZE):
```

Comment:
do we want to add similar tests to make sure the right number of tasks get called?